File: Mathematics/AllDivisorsofNumber.java
public class AllDivisorsofNumber {
    
    static void printDivisor(int n){
        int i;
        for(i=1; i*i<=n; i++){
            if(n%i==0)
                System.out.print(i+" ");
        }
        for( ;i>=1; i--){
            if(n%i==0)
                System.out.print(n/i +" ");
        }
    }
    public static void main(String[] args) {
        int number = 15;
         printDivisor(number);
    }
}

File: Mathematics/AllDivisorsofNumber.cpp
#include<iostream>
using namespace std;

//Naive Solution
 /*void printDivisors(int n){
    for(int i=1; i<=n; i++)
        if(n%i==0)
            cout<<i<<" ";
 }*/
 //Efficient Solution
 void printDivisors(int n){
    int i;
    for(i=1; i*i<=n; i++)
        if(n%i==0)
            cout<<i<<" ";
    for( ; i>=1; i--){
        if(n%i==0)
            cout<<n/i<<" ";
    }
 }
int main(){
    int number = 15;
    printDivisors(number);

}